lib/Workflow/Persister/DBI.pm

```diff
@@ -374,13 +391,13 @@ sub fetch_history {
     while ( my $row = $sth->fetchrow_arrayref ) {
         $self->log->debug("Fetched history object '$row->[0]'");
         push @history, {
-            id          => $row->[0],
-            workflow_id => $row->[1],
-            action      => $row->[2],
-            description => $row->[3],
-            state       => $row->[4],
-            user        => $row->[5],
-            date        => $self->parser->parse_datetime( $row->[6] ),
+            id          => $row->[$WORKFLOW_HIST_ID_FIELD_POS],
+            workflow_id => $row->[$WORKFLOW_ID_HIST_FIELD_POS],
+            action      => $row->[$WORKFLOW_ACTION_HIST_FIELD_POS],
+            description => $row->[$WORKFLOW_DESC_HIST_FIELD_POS],
+            state       => $row->[$WORKFLOW_STATE_HIST_FIELD_POS],
+            user        => $row->[$WORKFLOW_USER_HIST_FIELD_POS],
+            date        => $self->parser->parse_datetime( $row->[$WORKFLOW_DATE_HIST_FIELD_POS] ),
         };
```

Comment:
would it make sense to, instead, do:

```suggestion
    my %h;
    @h{qw(id workflow_id action description state user date)} = @{$row};
    push @history, \%h;
```

Which is supported on 5.14 and later (I verified 5.14; no idea which version introduces this).

Reply:
Something similar could be used in a number of other locations where the array content seems to be enumerated. On the other hand, maybe we can use the hashref function instead of the arrayref in order to get the fields from the query in a hash directly instead of enumerating the columns?